app/response_processor.py

```diff
@@ -80,6 +82,18 @@ def _send_receipt(self, decrypted, data):
                 res_logger.error("Receipt rejected by endpoint")
                 raise BadMessageError("Failure to send receipt")
 
+            elif res.status_code == 404:
+                namespaces = {'error': 'http://ns.ons.gov.uk/namespaces/resources/error'}
+                tree = etree.fromstring(res.content)
+                element = tree.find('error:message', namespaces).text
+
+                if '1009' in element:
```

Comment:
Yes. Now splitting string on `-`. The error code *should* always be the first element in the subsequent list. If error messages are changed downstream, this could break,